app/src/main/java/net/gaast/giggity/Schedule.java

```diff
@@ -421,6 +424,170 @@ private void loadIcal(BufferedReader in) {
+					JSONObject link = linklist.getJSONObject(i);
+					Schedule.Link slink = new Link(link.getString("link"), link.getString("name"));
+					slink.setType(link.optString("type", null));
+					links.addLast(slink);
+				}
+			}
+
+
+			if (conference.has("microlocations")) {
+
+                /*Changing the flag after checking the organizer is using with microlocations
+				options enabled*/
+
+				hasMicrolocs = true;
+
+				//Getting microlocations to add latitude and longitude
+				JSONArray microlocations = conference.getJSONArray("microlocations");
+
+				for (i = 0; i < microlocations.length(); i++) {
+
+					JSONObject room = microlocations.getJSONObject(i);
+					locs.put(room.getString("name"), room.getString("longitude") + "," + room.getString("latitude"));
+				}
+			}
+
+
+			//The sessions are contained in the array present in an object
+			JSONArray events = conference.getJSONArray("sessions");
+
+			for (i = 0; i < events.length(); i++) {
+
+				JSONObject event = events.getJSONObject(i);
+				String uid = event.getString("id");
+				String title = event.getString("title");
+
+                /*Our date format is different and I changed getTimeInMillis() a bit to ignore "+08"
+				in second part to avoid error in integer parsing*/
+				String startTimeS = event.getString("start_time");
+				String endTimeS = event.getString("end_time");
+				Date startTime, endTime;
+				startTime = new Date(getTimeInMillis(startTimeS));
+				endTime = new Date(getTimeInMillis(endTimeS));
+				Schedule.Item item = new Schedule.Item(uid, title, startTime, endTime);
+				item.setDescription(event.getString("long_abstract"));
+
+				if (event.getString("signup_url") != "null") {
+					item.addLink(new Link(event.getString("signup_url")));
+				}
+
+
+				JSONObject microlocation = event.getJSONObject("microlocation");
+				String location = microlocation.getString("name");
+
+				if ((line = tentMap.get(location)) == null) {
+					line = new Schedule.Line(location, location);
+					tents.add(line);
+					tentMap.put(location, line);
+				}
+
+
+				//Getting value (latitude and longitude) from the map by key (name)
+
+				if (hasMicrolocs) {
+					String locString = locs.get(line.getTitle());
+					String latitude = locString.substring(0, locString.indexOf(','));
+					String longitude = locString.substring(locString.indexOf(',') + 1);
+
+					//Adding location details here
+					String latlon = null;
+					try {
+						latlon = ("geo:0,0?q=" + longitude + "," +
+								latitude + "(" +
+								URLEncoder.encode(line.getTitle(), "utf-8") + ")");
+					} catch (UnsupportedEncodingException e) {
+						e.printStackTrace();
+					}
+
+
+					if (latlon != null) {
+						line.setLocation(latlon);
+					}
+				}
+
+				line.addItem(item);
+			}
+
+		} catch (JSONException e) {
+			Log.w("JSON parsing", " Unable to parse ", e);
+			e.printStackTrace();
+		}
+	}
+
+	private long getTimeInMillis(String timeString) {
```

Comment:
Thanks for looking up. I need to make few more changes as our format is being updated. Here you can read more about new format https://open-event-api.herokuapp.com/ , I guess this will be helpful for readme.